File: django-project/myproject/myapp/applications/util/code/subtitle_type.py
import unittest
from enum import Enum


class SubtitleType(Enum):
    AUTOMATIC = 0
    MANUAL = 1

    def to_string(self):
        if self == SubtitleType.AUTOMATIC:
            return "automatic_captions"
        elif self == SubtitleType.MANUAL:
            return "subtitles"


class TestSubtitleType(unittest.TestCase):
    def test_to_string(self):
        auto_string = SubtitleType.AUTOMATIC.to_string()
        self.assertEqual(auto_string, "automatic_captions")

        manual_string = SubtitleType.MANUAL.to_string()
        self.assertEqual(manual_string, "subtitles")


if __name__ == "__main__":
    unittest.main()

File: vue-project/my-vue-app/src/enums/subtitle-type.ts
enum SubtitleType {
  AUTOMATIC = 0, // 自動字幕
  MANUAL = 1 // 手動字幕
}

const SubtitleTypeLabel: { name: string; value: number }[] = [
  { name: '自動字幕', value: SubtitleType.AUTOMATIC },
  {
    name: '手動字幕',
    value: SubtitleType.MANUAL
  }
]

export { SubtitleType, SubtitleTypeLabel }
